Source-Code/FlareDraw.cs

```diff
@@ -192,9 +192,22 @@ private void GenerateBodyFlares()
             //ListChildren(sm.systemPrefab.rootBody, 0);
             //--- HACK--
 
+            //If Kerbin is parented to the Sun, set its SMA- otherwise iterate
+            //through celestial bodies to locate which is parented to the Sun
+            //and has Kerbin as a child. Set the highest parents SMA to kerbinSMA
             if(BodyFlare.kerbinSMA <= 0.0)
             {
-                BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                if(FlightGlobals.Bodies[1].referenceBody == FlightGlobals.Bodies[0])
+                    BodyFlare.kerbinSMA = FlightGlobals.Bodies[1].orbit.semiMajorAxis;
+                else
+                {                    
+                    foreach(CelestialBody current in FlightGlobals.Bodies)
```

Comment:
This looks fine. Is there a reason you couldn't check the .referenceBody field directly (that is, in the else block get Bodies[1].referenceBody, and verify its parent is the sun, and use its .orbit field)?  That'd be more efficient, assuming there's not something that prevents it from working that I'm overlooking.